### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by Beachman4**

### CODE REVIEW

File: ci/tests/test-multi/tests/src/browser/setBasicAuth.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const Flagpole = require("../../../dist/index.js").Flagpole;
+
+const suite = Flagpole.suite("Set Basic Authentication for browser tests").base(
+  "https://www.milesplit.info/"
+);
+
+suite
+  .browser("Homepage Loads")
+  .setBasicAuth({ username: "flocasts", password: "florocks!!!" })
```
Comment (karlsnyder0) [inline ci/tests/test-multi/tests/src/browser/setBasicAuth.ts:9]: We should avoid Flo stuff in this repo.

Comment (johnsickels) [reply ci/tests/test-multi/tests/src/browser/setBasicAuth.ts:9]: I agree, I updated my examples repo with non-Flo suites, feel free to borrow these!

https://github.com/johnsickels/testing-examples/tree/master/tests/src

Comment (Beachman4) [reply ci/tests/test-multi/tests/src/browser/setBasicAuth.ts:9]: whoops, I didn't even notice that. I just copied some existing tests over when I was working on this

**Comment by jasonbyrne** [comment]
What is holding this up?

**Comment by Beachman4** [comment]
> What is holding this up?

Not a lot of time to work on this lately, plan on finishing this up this weekend

**Comment by Beachman4** [comment]
> Can you help me understand the reason for `test-base/`, `test-javascript/`, `test-multi/`? Why not use the tests in the existing `./tests/` directory?

You know I'm trying to remember. I know there was a reason. I think it's the way I set up the tests.

I used matrixes to run the different test suites against different node versions. Lemme look at refactoring this to use just the built in tests.

**Comment by johnsickels** [comment]
Looks a lot better! Thank you. I have a PR in https://github.com/flagpolejs/flagpole/pull/158 this fixes some broken tests, all should pass now

**Comment by johnsickels** [comment]
@Beachman4 All tests should pass now, I reran the workflow, but looks stuck on old code. Can you push a fresh commit with master merged?

**Review by johnsickels — APPROVED** [review]
Awesome work! So excited

**Comment by Beachman4** [comment]
> Awesome work! So excited

Time to SEND IT

**Pull request merged**